### PR TITLE
pythonPackages.py-multihash: init at 0.2.3

### PR DESCRIPTION
**Pull request by Rakesh4G**
###### Motivation for this change
pythonPackages.py-multihash: init at 0.2.3

**Depends On:**
https://github.com/NixOS/nixpkgs/pull/83202
https://github.com/NixOS/nixpkgs/pull/83200
https://github.com/NixOS/nixpkgs/pull/83197

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/py-multihash/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  propagatedBuildInputs = [
+    pytestrunner
+    morphys
+    base58
+    six
+    variants
+    varint
+    pytestcov
+  ];
```
Comment (jonringer) [inline pkgs/development/python-modules/py-multihash/default.nix]:
```suggestion
  nativeBuildInputs = [
      pytestrunner
  ];
  
  propagatedBuildInputs = [
    morphys
    base58
    six
    variants
    varint

  ];
  
  checkInputs = [
      pytestcov
   ];
```

**Review by jonringer — COMMENTED** [review]
if upstream correctly declared the right package metadata

File: pkgs/development/python-modules/py-multihash/default.nix (hunk 1 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/py-multihash/default.nix]: done.

**Comment by jonringer** [comment]
please see the reviews of the other two pr's, they are still blocking this PR

**Comment by Rakesh4G** [comment]
> please see the reviews of the other two pr's, they are still blocking this PR

ok. Thanks.

File: pkgs/development/python-modules/py-multihash/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  propagatedBuildInputs = [
+    base58
+    morphys
+    six
+    variants
+    varint
+  ];
+
+  checkInputs = [
```
Comment (jonringer) [inline pkgs/development/python-modules/py-multihash/default.nix:33]:
```suggestion
  checkInputs = [
    pytest
```

```
  Download error on https://pypi.org/simple/: [Errno -2] Name or service not known -- Some packages may not be found!
  No local packages or working download links found for pytest
  error: Could not find suitable distribution for Requirement.parse('pytest')
```

File: pkgs/development/python-modules/py-multihash/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{ base58
+, buildPythonPackage
+, fetchPypi
+, lib
+, morphys
```
Comment (jonringer) [inline pkgs/development/python-modules/py-multihash/default.nix:5]:
```suggestion
, morphys
, pytest
```

File: pkgs/development/python-modules/py-multihash/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    morphys
+    six
+    variants
+    varint
+  ];
+
+  checkInputs = [
+    pytestcov
+  ];
+
```
Comment (jonringer) [inline pkgs/development/python-modules/py-multihash/default.nix:36]: whitespace
```suggestion

```

File: pkgs/development/python-modules/py-multihash/default.nix (hunk 2 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/py-multihash/default.nix:34]: Done

File: pkgs/development/python-modules/py-multihash/default.nix (hunk 4 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/py-multihash/default.nix]: done

File: pkgs/development/python-modules/py-multihash/default.nix (hunk 3 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/py-multihash/default.nix:5]: Done

File: pkgs/development/python-modules/py-multihash/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+, pytest
+, pytestcov
+, pytestrunner
+, six
+, variants
+, varint
+}:
+buildPythonPackage rec {
+  pname = "py-multihash";
+  version = "0.2.3";
```
Comment (jonringer) [inline pkgs/development/python-modules/py-multihash/default.nix:15]: I didn't investigate this very thoroughly, but it looks like python2 is unable to import any of the packages
```suggestion
  version = "0.2.3";
  disabled = isPy27;
```

```
builder for '/nix/store/hxf2k0jvfnz3gqqz1gsrnjhgn0l7z6rj-python2.7-py-multihash-0.2.3.drv' failed with exit code 2; last 10 log lines:
  Traceback:
  tests/test_multihash.py:11: in <module>
      from multihash import (
  multihash/__init__.py:9: in <module>
      from .multihash import (
  multihash/multihash.py:9: in <module>
      import multihash.constants as constants
  E   ImportError: No module named constants
  !!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
  =========================== 1 error in 0.17 seconds ============================
cannot build derivation '/nix/store/92nbj7r7mkz9872mnbxpi9ziy23vr7ys-env.drv': 1 dependencies couldn't be built
[0 built (1 failed)]
error: build of '/nix/store/92nbj7r7mkz9872mnbxpi9ziy23vr7ys-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/83203
1 package failed to build:
python27Packages.py-multihash

2 package built:
python37Packages.py-multihash python38Packages.py-multihash
```

File: pkgs/development/python-modules/py-multihash/default.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
+{ base58
+, buildPythonPackage
+, fetchPypi
```
Comment (jonringer) [inline pkgs/development/python-modules/py-multihash/default.nix:3]:
```suggestion
, fetchPypi
, isPy27
```

File: pkgs/development/python-modules/py-multihash/default.nix [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+, lib
+, morphys
+, pytest
+, pytestcov
+, pytestrunner
+, six
+, variants
+, varint
+}:
```
Comment (jonringer) [inline pkgs/development/python-modules/py-multihash/default.nix:12]: NIT
```suggestion
}:

```

File: pkgs/development/python-modules/py-multihash/default.nix [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    pytestcov
+  ];
+
+  meta = with lib; {
+    description = "Self describing hashes - for future proofing";
+    homepage = "https://github.com/multiformats/py-multihash";
+    license = licenses.mit;
+    maintainers = with maintainers; [ rakesh4g ];
+  };
+
```
Comment (jonringer) [inline pkgs/development/python-modules/py-multihash/default.nix:45]: NIT
```suggestion
```

Comment (Rakesh4G) [reply pkgs/development/python-modules/py-multihash/default.nix]: done

File: pkgs/development/python-modules/py-multihash/default.nix (hunk 7 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/py-multihash/default.nix:13]: done

File: pkgs/development/python-modules/py-multihash/default.nix (hunk 5 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/py-multihash/default.nix:17]: done.

**Comment by CMCDragonkai** [comment]
Is this ready to be merged?

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 83203 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package failed to build:</summary>  - python27Packages.py-multihash</details><details>  <summary>2 package built:</summary>  - python37Packages.py-multihash  - python38Packages.py-multihash</details>

**Pull request merged**